core/src/main/java/org/infinispan/interceptors/impl/EntryWrappingInterceptor.java

```diff
@@ -87,20 +88,65 @@
    private StateTransferLock stateTransferLock;
    private XSiteStateConsumer xSiteStateConsumer;
    private GroupManager groupManager;
+   private CacheNotifier notifier;
 
    private static final Log log = LogFactory.getLog(EntryWrappingInterceptor.class);
    private static final boolean trace = log.isTraceEnabled();
    private static final EnumSet<Flag> EVICT_FLAGS =
          EnumSet.of(Flag.SKIP_OWNERSHIP_CHECK, Flag.CACHE_MODE_LOCAL);
 
+   private final ReturnHandler dataReadReturnHandler = new ReturnHandler() {
```

Comment:
I've changed the "cached" ReturnHandlers to be lambdas and they're consistently about 1% slower than anonymous inner classes in the local read-only test. The generated assembly seems to be about the same, so the difference is probably some extra indirection when a lambda is called. It would almost certainly disappear if I get the JIT to inline the `ReturnHandlers` in `invokeInterceptorsSync`, but for now I'm keeping the inner classes.